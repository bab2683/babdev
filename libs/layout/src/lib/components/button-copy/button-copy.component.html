<button #button babdevCopy [valueToCopy]="value" (click)="removeFocus()">
  <fa-icon [icon]="icon"></fa-icon>
</button>
