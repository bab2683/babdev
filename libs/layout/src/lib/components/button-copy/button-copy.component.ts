import { Component, ElementRef, ViewChild } from '@angular/core';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { faCopy } from '@fortawesome/free-regular-svg-icons';

@Component({
  selector: 'babdev-button-copy',
  styleUrls: ['./button-copy.component.scss'],
  templateUrl: './button-copy.component.html'
})
export class ButtonCopyComponent {
  public icon: IconDefinition = faCopy;
  public value: string;

  @ViewChild('button', { static: true }) public button: ElementRef;

  public removeFocus(): void {
    this.button.nativeElement.blur();
  }
}
