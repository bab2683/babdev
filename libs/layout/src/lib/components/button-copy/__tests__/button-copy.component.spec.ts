import { NO_ERRORS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { HTMLElementMock } from '@babdev/testing';

import { ButtonCopyComponent } from '../button-copy.component';

describe('ButtonCopyComponent', () => {
  let component: ButtonCopyComponent;
  let fixture: ComponentFixture<ButtonCopyComponent>;
  let blurSpy: any;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ButtonCopyComponent, HTMLElementMock],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ButtonCopyComponent);
    component = fixture.componentInstance;

    component.button = TestBed.createComponent<any>(
      HTMLElementMock
    ).componentInstance;

    blurSpy = jest.spyOn(component.button.nativeElement, 'blur');
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should remove the focus from element', () => {
    component.removeFocus();
    expect(blurSpy).toHaveBeenCalled();
  });
});
