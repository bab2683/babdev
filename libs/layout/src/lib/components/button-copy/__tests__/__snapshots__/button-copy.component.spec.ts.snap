// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCopyComponent should match snapshot 1`] = `
<babdev-button-copy
  button={[Function HTMLElementMock]}
  icon={[Function Object]}
>
  <button
    babdevcopy=""
  >
    <fa-icon />
  </button>
</babdev-button-copy>
`;
